server/init.go

```diff
@@ -0,0 +1,182 @@
+package server
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+
+	"github.com/spf13/cobra"
+
+	crypto "github.com/tendermint/go-crypto"
+	"github.com/tendermint/go-crypto/keys"
+	"github.com/tendermint/go-crypto/keys/words"
+	cmn "github.com/tendermint/tmlibs/common"
+	dbm "github.com/tendermint/tmlibs/db"
+	"github.com/tendermint/tmlibs/log"
+
+	tcmd "github.com/tendermint/tendermint/cmd/tendermint/commands"
+	cfg "github.com/tendermint/tendermint/config"
+	tmtypes "github.com/tendermint/tendermint/types"
+)
+
+// InitCmd will initialize all files for tendermint,
+// along with proper app_options.
+// The application can pass in a function to generate
+// proper options. And may want to use GenerateCoinKey
+// to create default account(s).
+func InitCmd(gen GenOptions, logger log.Logger) *cobra.Command {
+	cmd := initCmd{
+		gen:    gen,
+		logger: logger,
+	}
+	return &cobra.Command{
+		Use:   "init",
+		Short: "Initialize genesis files",
+		RunE:  cmd.run,
+	}
+}
+
+// GenOptions can parse command-line and flag to
+// generate default app_options for the genesis file.
+// This is application-specific
+type GenOptions func(args []string) (json.RawMessage, error)
+
+// GenerateCoinKey returns the address of a public key,
+// along with the secret phrase to recover the private key.
+// You can give coins to this address and return the recovery
+// phrase to the user to access them.
+func GenerateCoinKey() (crypto.Address, string, error) {
+	// construct an in-memory key store
+	codec, err := words.LoadCodec("english")
+	if err != nil {
+		return nil, "", err
+	}
+	keybase := keys.New(
+		dbm.NewMemDB(),
+		codec,
+	)
+
+	// generate a private key, with recovery phrase
+	info, secret, err := keybase.Create("name", "pass", keys.AlgoEd25519)
```

Comment:
TODO: take an address like before instead.

Needs the keys stuff working

Reply:
Jae told me not to, in a call with me, him, and Rigel.
It was too complicated.

This is just for dev mode, real apps will hand-generated. So just make simplest start-up experience for testing.

Reply:
👍 Frey - In dev mode we will simply create it the address we need to use - otherwise pre-load address from genesis

Reply:
so to send transactions from this account, I will then use `--name name` and enter `pass` as the password ... just need to document it clearly